core/commonMain/src/implementations/immutableSet/TrieNode.kt

```diff
@@ -355,11 +385,9 @@ internal class TrieNode<E>(
         if (buffer[cellIndex] is TrieNode<*>) { // element may be in node
             val targetNode = nodeAtIndex(cellIndex)
             val newNode = targetNode.remove(elementHash, element, shift + LOG_MAX_BRANCHING_FACTOR)
-            return when {
-                targetNode === newNode -> this
-                newNode == null -> removeCellAtIndex(cellIndex, cellPositionMask)
-                else -> updateNodeAtIndex(cellIndex, newNode)
-            }
+            checkNotNull(newNode)
```

Comment:
Since we now cannot have a node with a single element anywhere in a trie except for the root node, we can rewrite `TrieNode.remove` to never return null and provide another path for removing the last remaining element from a map/set. However the effect of this change should be evaluated with the benchmarks.